lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -818,26 +819,147 @@ namespace Js
+            }
+
+            JavascriptString *currencyCodeJsString = JavascriptString::FromVar(propertyValue);
+            const char16 *currencyCode = currencyCodeJsString->GetSz();
+
+            // __currencyDisplayToUse: 0 is symbol, 1 is code, 2 is name.
+            if (GetTypedPropertyBuiltInFrom(options, __currencyDisplayToUse, TaggedInt))
+            {
+                uint16 currencyDisplay = TaggedInt::ToUInt16(propertyValue);
+                IfFailThrowHr(PlatformAgnostic::Intl::CreateCurrencyFormatter(locale, cch, currencyCode, currencyDisplay, &numberFormatter));
+            }
+        }
+        else
+        {
+            // Use the number formatter (0 or default)
+            IfFailThrowHr(PlatformAgnostic::Intl::CreateNumberFormatter(locale, cch, &numberFormatter));
+        }
+
+        Assert(numberFormatter);
+        // REVIEW (doilij): AutoPtr will call delete on IPlatformAgnosticResource and complain of non-virtual dtor. There are no IfFailThrowHr so is this still necessary?
+        // TODO (doilij): If necessary, introduce an PlatformAgnosticResourceAutoPtr that calls Cleanup() instead of delete on the pointer.
+        // PlatformAgnostic::Resource::AutoPtr<PlatformAgnostic::Resource::IPlatformAgnosticResource> numberFormatterGuard(numberFormatter);
+
+        // TODO (doilij): Render signed zero.
+
+        bool isDecimalPointAlwaysDisplayed = false;
+        bool useGrouping = true;
+
+        if (GetTypedPropertyBuiltInFrom(options, __isDecimalPointAlwaysDisplayed, JavascriptBoolean))
+        {
+            isDecimalPointAlwaysDisplayed = JavascriptBoolean::FromVar(propertyValue)->GetValue();
+        }
+
+        if (GetTypedPropertyBuiltInFrom(options, __useGrouping, JavascriptBoolean))
+        {
+            useGrouping = JavascriptBoolean::FromVar(propertyValue)->GetValue();
+        }
+
+        // Numeral system is in the locale and is therefore already set on the icu::NumberFormat
+        // TODO (doilij): extract the numbering system from the locale name (JS-side) and use that to set the __numberingSystem with fallback
+        // TODO (doilij): determine the numbering system for the ICU locale (so that works even if it wasn't specified directly)
+
+        // REVIEW (doilij): assuming the resolved language has already been set in __locale
+        // TODO (doilij): find out whether numberFormat->getLocale() has relevant extension tags for things like numeral system (-nu-)
+
+        if (HasPropertyBuiltInOn(options, __minimumSignificantDigits) || HasPropertyBuiltInOn(options, __maximumSignificantDigits))
+        {
+            // Do significant digit rounding
+            uint16 minSignificantDigits = 1, maxSignificantDigits = 21;
+
+            if (GetTypedPropertyBuiltInFrom(options, __minimumSignificantDigits, TaggedInt))
+            {
+                minSignificantDigits = max<uint16>(min<uint16>(TaggedInt::ToUInt16(propertyValue), 21), 1);
```

Comment:
Will this be impacted by https://github.com/Microsoft/ChakraCore/pull/3799 ?

Reply:
Yes, I believe so. I'll take note, thanks!